app/controllers/api/v2/filters_controller.rb

```diff
@@ -35,6 +35,14 @@ def show
 
       def create
         @filter = nested_obj ? nested_obj.filters.build(params[:filter]) : Filter.new(params[:filter])
+
+        User.current.roles.each do |user_role|
+          user_role.filters.each do |user_filter|
+            if user_filter.resource_type == @filter.resource_type
+              @filter.filters << user_filter
+            end
+          end
+        end
```

Comment:
If I understand the patch correctly I see several issues with this solution:

1) What happens if User.current is admin and he creates new filter? It may link some of his filters (which are ignored since he is admin) which does not seem to make sense.

2) Let's imagine we have user Alice with filter view_hosts with condition "name =~ a*" and she delegates view_hosts to Bob, later Eve removes the view_host filter from Alice so Alice can't see any host, Bob is now able to see all hosts because there's no limiting filter from Alice anymore

3) It's slightly better than copying all filters by linking other filters so updates of Alice filters propagates to Bob filters, but it still caches the current status when filter is being created. What if later admin grants Alice another filter with same permission view_hosts now with condition "name =~ b". Bob won't see such hosts, right? Maybe filters should not be linked to other filters but users that created them.

4) Easy to fix but serious issue, `User.current.roles` does not contain all user's roles, there are just directly assigned ones. You should use cached_roles to get roles granted by usergroup membership.

5) Inefficient and unreadable, how about this? (I'm sure it could be improved further)
```ruby
@filter.filters = User.current.filters.joins(:permissions).where('permissions.resource_type' => @filter.resource_type)
```